src/sentry/api/endpoints/organization_member_team_details.py

```diff
@@ -150,6 +187,43 @@ def post(
 
         return Response(serialize(team, request.user, TeamWithProjectsSerializer()), status=201)
 
+    def put(
+        self,
+        request: Request,
+        organization: Organization,
+        member: OrganizationMember,
+        team_slug: str,
+    ) -> Response:
+        try:
+            team = Team.objects.get(organization=organization, slug=team_slug)
+        except Team.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        try:
+            omt = OrganizationMemberTeam.objects.get(team=team, organizationmember=member)
+        except OrganizationMemberTeam.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        serializer = OrganizationMemberTeamSerializer(data=request.data, partial=True)
+        if not serializer.is_valid():
+            return Response(status=400)
+        result = serializer.validated_data
+
+        if "role" in result:
+            if features.has("organizations:team-roles", organization):
```

Comment:
Could combine these to save some nesting and make cleanup a 1 line change instead of > 1 lines.

Reply:
Sure. I want to say I've seen a soft convention where a feature flag check is in its own `if` statement, but I don't mind changing it.